### PR TITLE
Add startup library

### PR DESCRIPTION
**Pull request by bouguila**
this will fix Starcounter/HeadsOmni#299 once merged.

### CODE REVIEW

File: src/Search/ViewModels/AppName.json.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using Starcounter;
+using Starcounter.Startup.Routing;
+using Starcounter.Startup.Routing.Activation;
+
+namespace Search
+{
+    [Url("/search/app-name")]
+    partial class AppName : Json, IInitPageWithDependencies
+    {
+        public void Init() { }
```
Comment (joozek78) [inline src/Search/ViewModels/AppName.json.cs]: you don't have to implement `IInitPageWithDependencies` when you have no init body

File: src/Search/ViewModels/SearchAll.json.cs [hunk 2]
```diff
@@ -2,30 +2,45 @@
...
 {
-    partial class SearchAll : Json
+    [Url("/search/all?query={?}")]
+    partial class SearchAll : Json, IBound<string>, IInitPageWithDependencies
     {
-        static SearchAll()
+        public void Init()
         {
-            DefaultTemplate.Pagination.PageSize.DefaultValue = 100;
+            DefaultTemplate.Pagination.PageSize.DefaultValue = 20;
```
Comment (joozek78) [inline src/Search/ViewModels/SearchAll.json.cs]: this is initialization of static properties, it should stay in the static ctor

**Comment by bouguila** [comment]
ping @joozek78

File: src/Search/Search.csproj [hunk 3]
```diff
@@ -44,15 +44,46 @@
     <PlatformTarget>x64</PlatformTarget>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Microsoft.Extensions.Configuration.Abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\..\packages\Microsoft.Extensions.Configuration.Abstractions.2.0.0\lib\netstandard2.0\Microsoft.Extensions.Configuration.Abstractions.dll</HintPath>
```
Comment (joozek78) [inline src/Search/Search.csproj]: I missed it last time - please use [`$(SolutionDir)`](https://github.com/Starcounter/HeadsOmni/#when-adding-new-nuget-dependencies-to-sub-projects)

File: src/Search/Core/SearchHandler.cs [hunk 4]
```diff
@@ -5,11 +5,11 @@ namespace Search
 {
     public class SearchHandler
     {
-        protected MatchingProvider MatchingProvider;
+        protected IMatchingProvider matchingProvider;
 
         public SearchHandler()
         {
-            MatchingProvider = new MatchingProvider();
+            matchingProvider = new MatchingProvider();
```
Comment (joozek78) [inline src/Search/Core/SearchHandler.cs]: why not express this as a dependency?
```c#
public SearchHandler(IMatchingProvider matchingProvider)
{
	_matchingProvider = matchingProvider;
}
```

File: src/Search/Core/SearchHandler.cs [hunk 5]
```diff
@@ -5,11 +5,11 @@ namespace Search
 {
     public class SearchHandler
     {
-        protected MatchingProvider MatchingProvider;
+        protected IMatchingProvider matchingProvider;
```
Comment (joozek78) [inline src/Search/Core/SearchHandler.cs]: why is this protected? is someone deriving from this class?

File: src/Search/Models/SystemUser.cs [hunk 6]
```diff
@@ -1,4 +1,9 @@
-﻿using Starcounter;
+﻿using System;
```
Comment (joozek78) [inline src/Search/Models/SystemUser.cs]: why do you add those? If they weren't there before and it compiled then those usings are redundant

Comment (bouguila) [reply src/Search/Models/SystemUser.cs]: i just re-added them to not break the current mapping as the app wont start unless mappers are updated. but there is no actual need for them and i'm removing them

File: src/Search/Core/SearchHandler.cs (hunk 5 above)
Comment (bouguila) [reply src/Search/Core/SearchHandler.cs]: i have no idea why, i just kept the same modifiers.

File: src/Search/Core/SearchHandler.cs (hunk 4 above)
Comment (bouguila) [reply src/Search/Core/SearchHandler.cs]: 👍

File: src/Search/Search.csproj (hunk 3 above)
Comment (bouguila) [reply src/Search/Search.csproj]: my bad 🤦‍♂️

**Comment by bouguila** [comment]
@joozek78 could you please give it a final round ?

**Comment by bouguila** [comment]
thanks @joozek78 for the review.

**Pull request merged**